bin/reth-bench/src/bench/new_payload_only.rs

```diff
@@ -43,21 +43,25 @@ impl Command {
         let (sender, mut receiver) = tokio::sync::mpsc::channel(1000);
         tokio::task::spawn(async move {
             while benchmark_mode.contains(next_block) {
+                let fetch_start = Instant::now();
+
                 let block_res =
                     block_provider.get_block_by_number(next_block.into(), true.into()).await;
+                let fetch_duration = fetch_start.elapsed();
                 let block = block_res.unwrap().unwrap();
                 let block = from_any_rpc_block(block);
 
                 next_block += 1;
-                sender.send(block).await.unwrap();
+                sender.send((block, fetch_duration)).await.unwrap();
             }
         });
 
         // put results in a summary vec so they can be printed at the end
         let mut results = Vec::new();
         let total_benchmark_duration = Instant::now();
 
-        while let Some(block) = receiver.recv().await {
+        let mut total_fetch_duration = Duration::ZERO;
+        while let Some((block, fetch_duration)) = receiver.recv().await {
             // just put gas used here
             let gas_used = block.gas_used;
```

Comment:
given this is happening in a separate task, is this accurate? it seems like we might subtract too much

Reply:
can you pls elaborate? happens in a separate task but we record and return the `Duration` that the fetch operations per block took, in the other task we aggregate these durations and subtract them from the total benchmark time so far, is this correct?

Reply:
oh i see what you mean, the channel size is 1000, the block downloader task can progress without blocking the engine interaction task, it is not that easy to know if the block downloads are actually affecting the engine interactions

Reply:
> the block downloader task can progress without blocking the engine interaction task, it is not that easy to know if the block downloads are actually affecting the engine interactions

Yeah exactly, I think we would instead want to subtract any blocking that occurs in the main task, which currently happens in the loop condition (`await`ing on the `recv`)

Reply:
yep that's it, pushed the changes PTAL